addons/mail/static/src/views/form/form_renderer.js

```diff
@@ -0,0 +1,48 @@
+/** @odoo-module */
+
+import { useModels } from '@mail/component_hooks/use_models';
+import { ChatterContainer } from "@mail/components/chatter_container/chatter_container";
+import { WebClientViewAttachmentViewContainer } from '@mail/components/web_client_view_attachment_view_container/web_client_view_attachment_view_container';
+
+import { patch } from "@web/core/utils/patch";
+import { FormRenderer } from "@web/views/form/form_renderer";
+
+patch(FormRenderer.prototype, 'mail', {
```

Comment:
This patch only exists so that the AttachmentViewer can be rendered conditionally, correct? Can't this duty simply be delegated to the WebClientViewAttachmentViewContainer? I'd rather we avoid patches unless there is no other option

Reply:
It's necessary for the condition in `hasAttachmentViewer()`, which controls both the rendering of `WebClientViewAttachmentViewContainer`, but critically also plenty of other things (including props and the position of `ChatterContainer`).

The first common parent to all of these is the form renderer.